### PR TITLE
[FEAT]: Integrate Openai API in our backend

### PR DESCRIPTION
**Pull request by tumero58**

### CODE REVIEW

File: constants/openai.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export const OPEN_AI_URL:string = "https://api.openai.com/v1/completions";
+export const OPEN_AI_API_KEY:string = `${process.env.OPEN_AI_API_KEY}`;
+export const OPEN_AI_MODEL: string = "text-davinci-003";
+export const OPEN_AI_PROMPT_KEY: string = "open-ai-prompt";
```
Comment (hrachocode) [inline constants/openai.ts:4]: these 2 are static constants?

**Review by hrachocode — CHANGES_REQUESTED** [review]
I guess file name should be api/createPost for better understanding

**Comment by tumero58** [comment]
[OpenAI research](https://docs.google.com/document/d/e/2PACX-1vRvYYSViAQCBClNlEOvZ889kA7XiAodPQoBAfh_dPFKnfgo13XfqOnMA0MXEA0ucqsrH-MZG3Zg0Xno/pub)

File: constants/openai.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export const OPEN_AI_TEXT_URL:string = `${process.env.OPEN_AI_TEXT_URL}`;
+export const OPEN_AI_IMAGE_URL:string = `${process.env.OPEN_AI_IMAGE_URL}`;
+export const OPEN_AI_API_KEY:string = `${process.env.OPEN_AI_API_KEY}`;
+export const OPEN_AI_MODEL: string = "text-davinci-003";
+export const OPEN_AI_PROMPT_KEY: string = "text-prompt";
+export const OPEN_AI_IMAGE_PROMPT_KEY: string = "image-prompt";
+export const OPEN_AI_MAX_TOKENS: number = 4000;
+export const OPEN_AI_TEMPERATURE: number = 0.5;
+export const OPEN_AI_IMAGE_SIZE: string = "1024x1024";
```
Comment (hrachocode) [inline constants/openai.ts:9]: does this size fits with the size in design?

File: utils/openai/generatePost.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  
+  const textResult = await httpClient.post(`${process.env.NEXT_PUBLIC_APP_URL}${OPEN_AI_TEXT_ROUTE}`, {
+    [OPEN_AI_PROMPT_KEY]: textPrompt
+  });
+
+  const imageResult = await httpClient.post(`${process.env.NEXT_PUBLIC_APP_URL}${OPEN_AI_IMAGE_ROUTE}`, {
+    [OPEN_AI_IMAGE_PROMPT_KEY]: imagePromt
+  });
+
+  return { textResult, imageResult };
```
Comment (hrachocode) [inline utils/openai/generatePost.ts:15]: imageResult is a link or is a binary representation of the image?

**Review by hrachocode — CHANGES_REQUESTED** [review]

File: constants/openai.ts (hunk 2 above)
Comment (tumero58) [reply constants/openai.ts:9]: it is higher so it won't hurt the quality

File: utils/openai/generatePost.ts (hunk 3 above)
Comment (tumero58) [reply utils/openai/generatePost.ts:15]: it is an object with a image link inside

**Review by hrachocode — APPROVED** [review]

**Pull request merged**